rp2040-hal/examples/rtc_irq_example.rs

```diff
@@ -0,0 +1,176 @@
+        XTAL_FREQ_HZ,
+        pac.XOSC,
+        pac.CLOCKS,
+        pac.PLL_SYS,
+        pac.PLL_USB,
+        &mut pac.RESETS,
+        &mut watchdog,
+    )
+    .ok()
+    .unwrap();
+
+    // The single-cycle I/O block controls our GPIO pins
+    let sio = hal::Sio::new(pac.SIO);
+
+    // Set the pins to their default state
+    let pins = gpio::Pins::new(
+        pac.IO_BANK0,
+        pac.PADS_BANK0,
+        sio.gpio_bank0,
+        &mut pac.RESETS,
+    );
+
+    // Configure GPIO 25 as an output to drive our LED.
+    // we can use reconfigure() instead of into_pull_up_input()
+    // since the variable we're pushing it into has that type
+    let led = pins.gpio25.reconfigure();
+
+    // Prepare the RTC for the example using the 1/1/0 (Day/Month/Year) at 0:00:00 as the initial
+    // day and time (it may not have been a Monday but it doesn't matter for this example).
+    let mut rtc = hal::rtc::RealTimeClock::new(
+        pac.RTC,
+        clocks.rtc_clock,
+        &mut pac.RESETS,
+        rtc::DateTime {
+            year: 0,
+            month: 1,
+            day: 1,
+            day_of_week: rtc::DayOfWeek::Monday,
+            hour: 0,
+            minute: 0,
+            second: 0,
+        },
+    )
+    .unwrap();
+
+    // Trigger the IRQ every time a minute starts.
+    rtc.schedule_alarm(rtc::DateTimeFilter::default().second(0));
+    rtc.enable_interrupt();
+
+    // Give away our pin and rtc by moving them into the `GLOBAL_SHARED` variable.
+    // We won't need to access them in the main thread again
+    critical_section::with(|cs| {
+        GLOBAL_SHARED.borrow(cs).replace(Some((led, rtc)));
+    });
+
+    // Unmask the RTC IRQ so that the NVIC interrupt controller
+    // will jump to the interrupt function when the interrupt occurs.
+    // We do this last so that the interrupt can't go off while
+    // it is in the middle of being configured
+    unsafe {
+        pac::NVIC::unmask(pac::Interrupt::RTC_IRQ);
+    }
+
+    loop {
+        // interrupts handle everything else in this example.
+        cortex_m::asm::wfi();
+    }
+}
+
+#[allow(non_snake_case)]
+#[interrupt]
+fn RTC_IRQ() {
+    // The `#[interrupt]` attribute covertly converts this to `&'static mut Option<LedAndRtc>`
+    static mut LED_AND_RTC: Option<LedAndRtc> = None;
+
+    // This is one-time lazy initialisation. We steal the variables given to us
+    // via `GLOBAL_SHARED`.
+    if LED_AND_RTC.is_none() {
+        critical_section::with(|cs| {
+            *LED_AND_RTC = GLOBAL_SHARED.borrow(cs).take();
+        });
+    }
+
+    // Need to check if our Option<LedAndButtonPins> contains our pins
+    if let Some(led_and_rtc) = LED_AND_RTC {
+        // borrow led and rtc by *destructuring* the tuple
+        // these will be of type `&mut LedPin` and `&mut RealTimeClock`, so we don't have
+        // to move them back into the static after we use them
+        let (led, rtc) = led_and_rtc;
```

Comment:
I had to look up why this works. Even if I knew it in the past.
The answer is "default binding modes", https://doc.rust-lang.org/reference/patterns.html#binding-modes

Would it be useful to mention that here, or would it distract too much from the point of this example?

Reply:
I'll add a comment above the if-let